### DISCUSSION TITLE
L&G E450 Switzerland

### DISCUSSION THREAD

**Discussion by se4n01** [scs/smartmeter-datacollector#56] [General]
Success! I used: 
- Debian 11 (Bullseye)
- https://www.aliexpress.com/item/32894249052.html M-Bus slave
- I used a RJ12 cable using its middle pins (3&4) Mbus is polarity agnostic. 
- This can be useful: https://www.aliexpress.com/item/32868498701.html

Good luck! And thanks to the developer for this great software!

**Comment by mhin76** [comment]
@se4n01 good to know that you got it working even with an M-Bus slave. I also have bought one, different one (https://de.aliexpress.com/item/1005004540180026.html?spm=a2g0o.order_list.order_list_main.5.21ef5c5fkO3Ie3&gatewayAdapt=glo2deu) and just upgraded to Debian 11. But I am not getting any data to my MQTT :-( How often does your L&G send data? Do you have any LED on your MBUS-USB that indicates any data reading? To which MQTT topic does it post? besides installing smartmeter and configuring it via webpage there is nothing more to do, right? I see it does run as a process.
Thanks for your help.

**Reply by RetoWettstein** [reply]
@se4n01 Oh wow. Congratulation. I am trying the same in Switzerland. Sadly with no success right now. Did the provider had to activate the smartmeter to send data over Mbus? Or did your TX and RX LED on the Mbus to USB device flashes from time to time? The LED on my device are off all the time. Thank you so much for help.

**Reply by berwegera** [reply]
In my neighbourhood in Switzerland, the local electricity company also had to "do something" to get the data through. What they had to do is beyond my knowledge.

The data on my MQTT server looks like this:

smartmeter
 - xyzNummerSmartmeter
 - - ACTIVE_POWER_P = {"value": 88.0, "timestamp": 1694674094}
 - - ACTIVE_POWER_N = {"value": 34.0, "timestamp": 1694674094}
 - - ACTIVE_ENERGY_P = {"value": 1186679.0, "timestamp": 1694674094}
 - - ACTIVE_ENERGY_N = {"value": 3447220.0, "timestamp": 1694674094}
 - - REACTIVE_ENERGY_Q1 = {"value": 1305.0, "timestamp": 1694674094}
 - - REACTIVE_ENERGY_Q2 = {"value": 1976.0, "timestamp": 1694674094}
 - - REACTIVE_ENERGY_Q3 = {"value": 1306846.0, "timestamp": 1694674094}
 - - REACTIVE_ENERGY_Q4 = {"value": 1057299.0, "timestamp": 1694674094}
 - - ACTIVE_ENERGY_P_T1 = {"value": 256695.0, "timestamp": 1694674044}
 - - ACTIVE_ENERGY_P_T2 = {"value": 929984.0, "timestamp": 1694674044}

**Reply by mhin76** [reply]
interesting. our L&G in Staufen only has an IR interface and the MBus behind the coverage which is blocked (blombiert). However I saw that a cable connects the water meter which they can read automatically remotely. I measured the voltage on that cable and got about 27V. So without seeing the connection to L&G I am quite sure it connects to the MBus. I connected my MBUS-Slave/USB to the same cable (as it is a bus, you should be able to connect various), but so far nothing.
@berwegera what kind of MBus/USB adapter do you use? Is it a slave or master?

**Reply by berwegera** [reply]
Here is more or less the same installation, except that EW was kind enough to pull a cable from the MBus to the outside at my request when installing the new meter ... Thank you EW!  Then I connected the following adapter (slave! -> 12.-Fr) -> https://de.aliexpress.com/item/1005004334750447.html . Then it remained "dark" on the shell. Then they did some magic .... and yes .... then it worked.

Magic -> Excerpt of the e-mail from the electricity company: I have received some more input regarding the customer interface for the smart meter, which means I have to go to the meter again on site to adjust a few settings.

**Reply by mhin76** [reply]
thanks for your quick answer. on that link, you bought the thin adapter (Slave) and not the other one (Master), right?
Salve would make sense as the smartmeter sends data out at certain interval.
So one of the electric company had to change something on site to enable the customer interface? any idea if he used that plombed reset button?

**Reply by raymar9** [reply]
Usually the provider needs to enable the (MBus) interface on your smartmeter to enable the data pushing. In the case of EKZ, this data is even encrypted and you need to enter the key, EKZ is giving you, into the configuration of this tool to decrypt the received messages. I do not know how this is handled with other electricity providers. 

Our software only support listening to pushed data from the smartmeter. Request/response is not supported and I expect also not to be supported by the provider (security risk).

@mhin76  Only use MBus **Slave** adapters since one of the meters is already the MBus master and only one master is allowed in a MBus bus. Probably, this MBus connection between your meters is reserved for reading out the water meter and no push data service is provided there. But this is only an assumption - you should call your electricity provider.

**Reply by raymar9** [reply]
@mhin76 I expect them to use a specific software and the IR interface to configure your smart-meter on site. Or they are able to remotely configure the smartmeter over air or Powerline communiction (after all it is a "smart"meter ;) ).

**Reply by mhin76** [reply]
@raymar9 good points. they once also told me that if they activate the IR interface, the remote would not work anymore. And I thought so that salve is sufficient (found on internet several info with master). Also good point that the smart meter per default might not just send out those data regularly, so it might need to be enabled. I will further investigate.
Would you have any contact person at EKZ?

**Reply by raymar9** [reply]
@mhin76  You can call the normal support hotline. They are briefed on this topic and start the process. I do not know any contact person at EKZ who is responsible for setting up the smart meters.

**Comment by RetoWettstein** [comment]
Thank you @berwegera.
@mhin76 I heard, that it is allowed to cut the plomb for the Userinterface. I am in connection with my local eltricity company. They told me, to use the IR intercace is difficult, because it did not push the data. You have to send commands and then you will get data from IR interface. But there where diffrent security levels. So i decided to use the RJ12 interface. I use the Ali mbus usb converter as a slave on RaspberryPi. But it seems there where no data pushed yet. Wrote an email to my grid company again. Maybe they could help.  
Hope they can activate the function.

![IMG_4670](https://github.com/scs/smartmeter-datacollector/assets/56090565/ab1eabd5-09a2-456b-b422-2a4af17c3557)

**Reply by mhin76** [reply]
so there are two plombs. one for the cable connection, I read in some instructions of other Gemeinde that they allow to take off that plomb. My responsible here clearly indicates, that this needs to stay otherwise you could "steal" energy. The other plomb is on the rest button on the L&G to access the setting menu, where you can set MBus specific configuration. Might be that there something needs to be activated...
I tried IR before, without success.

**Reply by mhin76** [reply]
based on user manual there are 4 MBUS channels to be configured. I thought e.g. this MBUS-Slave/USB would listen to all, also communication with the water meter. But I guess this tool already filters those communication out. Do you see on your adapter any LED activities?
![image](https://github.com/scs/smartmeter-datacollector/assets/49210075/1eaa496e-72cb-44cc-ba3d-894bcc5952ce)

ah, just found the user manual of this MAP110 Service Tool. Seems that the MBUS can be configured as point-to-point or local multiple connection (see chapter 6.3.3)
http://nepa-ru.com/Landys+Gyr_files/services/01_web_map110_operating_en.pdf


Ich schreibe jetzt mal SWL an, denn scheint so, dass die doch noch was aktivieren müssen.

**Comment by RetoWettstein** [comment]
Thank you all for your help. Now i had success in Switzerland Kanton Aargau. My Provider had to activate the Smartmeter to push the data over the Mbus (remotely). NO decryption needed. I use the Ali slave Mbus usb stick. Connected to a raspberrypi with the latest image from EKZ. In IoBroker i get the Values over MQTT as a string. So had to convert it. 
![IMG_4675](https://github.com/scs/smartmeter-datacollector/assets/56090565/bf85828b-1a30-4557-ac02-547e05a44d4c)

**Reply by berwegera** [reply]
Congratulations Reto! Do you get the same values as me above, or do you get more? I would still be interested in one or two more values in your list :) and I'm just wondering how I can get them :)

**Reply by mhin76** [reply]
Gratulation!! Great success!! I am still in contact with SWL (Lenzburg) energy provider. My contact person so far is still puzzled what they have to do where. Hope they all have the same remote control software. Do you have a contact person at your energy provider? Maybe I could connect them :) My L&G 450 does not have this external plug, but they have connected MBUS wire already to read my water meter. As the MBUS is a bus I should just be able to connect my adapter to it and it should work.

Does any LED on your MBus/USB stick light up without MBUS connection? I wonder if my device https://de.aliexpress.com/item/1005004540180026.html?spm=a2g0o.order_list.order_list_main.16.21ef5c5fE2Awi0&gatewayAdapt=glo2deu is working at all.

**Reply by RetoWettstein** [reply]
@berwegera Thx. I get the following values via MQTT in IoBroker:
What Values are you interested in?
![SmartmeterPushedData](https://github.com/scs/smartmeter-datacollector/assets/56090565/0aeff640-6b23-4e30-ab5d-f6c06c131619)

**Reply by RetoWettstein** [reply]
@mhin76 Thank you. I saw pics in the web where the L&G450 did not have the RJ12 Plug. But i think under the case there are the Mbus connectors. But its not allowed to open them. But there is a new law in Switzerland and you have the right to collect and have access to your own data. Maybe an SWL technician has to mount an external Plug for you.
And yes, my RX Led flashes on the Mbus module. All 5 seconds after the provider activated the user interface remotely.
Before no RX and TX Leds flashed. Only the power LED. But think this is the power from the USB port.
My provider told me that in the near future they will increase the security level and they will activate the encryption.
So we need the encryption-Key from the provider.
i am not shure if the userinterface and the watermeter are on the same Mbus. sry
hmm i have a contact but hope you understand that i didn't want to share on github.
But Lenzburg is a big city. They should know how to unlock your interface ;)
good luck
![450w](https://github.com/scs/smartmeter-datacollector/assets/56090565/cf8683b7-d702-4cc6-883b-cfcc654b5035)
![450wo](https://github.com/scs/smartmeter-datacollector/assets/56090565/88e0bc5d-c965-49fd-a22e-ce7e388ebeaa)

**Reply by RetoWettstein** [reply]
@mhin76 https://forum.iobroker.net/topic/54099/smartmeter-wiener-netze-auslesen

**Reply by mhin76** [reply]
Thanks @RetoWettstein so far SWL told me that the technische Kommission from Staufen needs to first give the ok on activating the interface. I think they have just no clue. But apparently they only install them. AEW is the company administering it. Will reach out to them. Would you have any link to this swiss law?
About the MBus, as seen on the pic of our link it is pin 28 and 29. They are available twice, but to my knowledge exactly the same wiring. Means I should be able to use the one of the water meter. But my power led on my mbus (think one indicates power) is also off. I ordered another adapter that others used in this forum. I will keep you posted :)

**Reply by massimo1971** [reply]
Hi Reto, could you explain me how I can convert the String? I need the value aslo in ioBroker.

**Reply by RetoWettstein** [reply]
hi massimo, maybe there is a better way to do it but this script works for me.
![grafik](https://github.com/scs/smartmeter-datacollector/assets/56090565/f75c9444-7528-4f0d-af44-3305a53dc371)

**Reply by massimo1971** [reply]
Hi Reto, many thanks. I just ordered the same Mbus modul as you have posted above, Until now i tried the Mbus plug from gPlug.ch , which is much comfortable. but the AmsReader Software can not distingue from dayrate / nightrate. That's the reason why I would try data-collector.

**Comment by se4n01** [comment]
@RetoWettstein congrats on the success, sorry for my slow response I was on vacation until now. 

@mhin76 there are no lights on my mbus slave, the frequency the meter is sending data is quite high, every few s. I had to check the right wires were connected on the RJ interface, as soon as they were I saw the messages coming through within seconds. FWIW I also tried the IR interface but I had no luck with it. Let us know how it goes with the new adapter, when it is working it is very pleasing!

My best,
Sean

**Comment by RetoWettstein** [comment]
Hi massimoI made a Blockly Script to extract the plain Value and then convert it to an Integer. If you want i can send you the script or the picture from Blockly in the evening. Am 03.03.2024 um 14:08 schrieb massimo1971 ***@***.***>:﻿
Hi Reto, could you explain me how I can convert the String? I need the value aslo in ioBroker.
